api/envoy/config/cluster/v3/cluster.proto

```diff
@@ -541,6 +541,34 @@ message Cluster {
     google.protobuf.Duration max_interval = 2 [(validate.rules).duration = {gt {nanos: 1000000}}];
   }
 
+  // [#not-implemented-hide:]
+  message PrefetchPolicy {
+    // Indicates how many many requests can be anticipated per-upstream for each
+    // request, useful for high-QPS or latency-sensitive services.
+    //
+    // For example if this is 2, for an incoming HTTP/1.1 request, 2 connections will be
+    // established, one for the new incoming request, and one for a presumed follow-up request. For
+    // HTTP/2, only one connection would be established by default as one connection can
+    // serve both the original and presumed follow-up request.
+    //
+    // In steady state for non-multiplexed connections a value of 1.5 would mean if there were 100
+    // active requests, there would be 100 connections in use, and 50 connections prefetched.
+    // This might be a useful value for something like short lived single-use connections,
+    // for example proxying HTTP/1.1 if keep-alive were false and each request resulted in connection
+    // termination. It would likely be overkill for long lived connections, such as TCP proxying SMTP
+    // or regular HTTP/1.1 with keep-alive. For long lived traffic, a value of 1.05 would be more
+    // reasonable, where for every 100 connections, 5 prefetched connections would be in the queue
+    // in case of unexpected disconnects where the connection could not be reused.
+    //
+    // If this value one, Envoy will fetch as many connections as needed to serve requests in
+    // flight. This means in steady state if a connection is torn down, a subsequent requests will
+    // pay an upstream-rtt latency penalty waiting for requests to be prefetched.
+    //
+    // This is limited somewhat arbitrarily to 3 because prefetching connections too aggressively can
+    // harm latency more than the prefetching helps.
```

Comment:
Per below the way this is written it will be forced to be set, there is no default. I would either change that or clarify above that the field must be set between 1.0-3.0 inclusive.